### ISSUE TITLE
websocket: close 1006 (abnormal closure): unexpected EOF

### ISSUE THREAD

**Issue by JacobSteelsmith** [livekit/egress#101]
Thanks again for the hard work here. I've deployed livekit and livekit egress on AWS EKS using helm. I'm recording MP4 and uploading to S3. Everything seems to be working well except when I stop the recording using stopEgress(), "websocket: close 1006 (abnormal closure): unexpected EOF" is logged and then it takes several minutes for the recording to show up in S3. 

Previously the recording was uploaded right after we disconnected from livekit. At that time we had not implemented stopEgress(). We get the same error with disconnect except for one out of around 20 tests. That one time there was no error and it uploaded right away. 

We had made some changes to the firewall as well as replaced some of the clusters when the error started. Since then, I've updated to the latest livekit server, egress, and client and server JS APIs but the scenario persists. I've reviewed the documentation and firewall settings but am unable to find the issue. I'm presuming it's something I did with our configuration but cannot find it. Our firewall rules match the documentation and the connection tester is all green without any warnings. The recordings look good once they reach S3 and are high quality, and the system is stable. We could use it as is but I'm presuming the error isn't optimal and it's nearly a requirement for us for the recordings to be available sooner. 

I'm not seeing anything of concern on the livekit server. Here is the full log from egress:

```{"level":"info","ts":1662672670.2748249,"logger":"egress","caller":"service/handler.go:129","msg":"egress updated","nodeID":"NE_Mk9becqXtdx3","egressID":"EG_px2xQzW8J7r7","status":"EGRESS_ACTIVE"}
{"level":"debug","ts":1662672671.2679987,"logger":"egress","caller":"service/handler.go:81","msg":"handling request","nodeID":"NE_Mk9becqXtdx3","egressID":"EG_px2xQzW8J7r7","requestID":"RPC_UZU79tyCFQQ7"}
{"level":"info","ts":1662672671.2680428,"logger":"egress","caller":"service/handler.go:129","msg":"egress updated","nodeID":"NE_Mk9becqXtdx3","egressID":"EG_px2xQzW8J7r7","status":"EGRESS_ENDING"}
{"level":"debug","ts":1662672671.268796,"logger":"egress","caller":"service/handler.go:147","msg":"request handled","nodeID":"NE_Mk9becqXtdx3","egressID":"EG_px2xQzW8J7r7","requestID":"RPC_UZU79tyCFQQ7","senderID":"ND_XjdjZkw9"}
{"level":"debug","ts":1662672671.2688847,"logger":"egress","caller":"pipeline/pipeline.go:451","msg":"sending EOS to pipeline","nodeID":"NE_Mk9becqXtdx3","egressID":"EG_px2xQzW8J7r7"}
{"level":"debug","ts":1662672671.2689269,"logger":"egress","caller":"source/appwriter.go:507","msg":"draining","nodeID":"NE_Mk9becqXtdx3","egressID":"EG_px2xQzW8J7r7","trackID":"TR_VCJ4H38uFva7Wp","kind":"video"}
{"level":"debug","ts":1662672671.2689493,"logger":"egress","caller":"source/appwriter.go:507","msg":"draining","nodeID":"NE_Mk9becqXtdx3","egressID":"EG_px2xQzW8J7r7","trackID":"TR_AMURL74WvcV3i2","kind":"audio"}
{"level":"debug","ts":1662672671.486058,"logger":"egress","caller":"pipeline/pipeline.go:268","msg":"EOS received, stopping pipeline","nodeID":"NE_Mk9becqXtdx3","egressID":"EG_px2xQzW8J7r7"}
{"level":"debug","ts":1662672671.488385,"logger":"egress","caller":"pipeline/pipeline.go:579","msg":"pipeline stopped","nodeID":"NE_Mk9becqXtdx3","egressID":"EG_px2xQzW8J7r7"}

Error:
{"level":"error","ts":1662672671.4932966,"logger":"egress","caller":"runtime/asm_amd64.s:1571","msg":"error with read worker","nodeID":"NE_Mk9becqXtdx3","error":"websocket: close 1006 (abnormal closure): unexpected EOF","stacktrace":"runtime.goexit\n\t/usr/lib/go-1.18/src/runtime/asm_amd64.s:1571"}

Then 258 seconds later:
{"level":"debug","ts":1662672929.3446214,"logger":"egress","caller":"pipeline/pipeline.go:606","msg":"uploading to s3","nodeID":"NE_Mk9becqXtdx3","egressID":"EG_px2xQzW8J7r7"}
{"level":"debug","ts":1662672929.4825525,"logger":"egress","caller":"pipeline/pipeline.go:639","msg":"removing temporary directory","nodeID":"NE_Mk9becqXtdx3","egressID":"EG_px2xQzW8J7r7","path":"/EG_px2xQzW8J7r7/"}
{"level":"info","ts":1662672929.48277,"logger":"egress","caller":"service/handler.go:127","msg":"egress completed","nodeID":"NE_Mk9becqXtdx3","egressID":"EG_px2xQzW8J7r7"}
{"level":"info","ts":1662672929.4856944,"logger":"egress","caller":"service/service.go:265","msg":"deleting handler temporary directory","nodeID":"NE_t2FZZB73NKJf","path":"/tmp/lkegress/EG_px2xQzW8J7r7"}
```

**Comment by frostbyte73** [comment]
To reproduce this, do I just need to start a track composite, and then have the participant being recorded leave the room?

**Comment by JacobSteelsmith** [comment]
Yes, we use track composite. It happens with us either leaving the room by calling Room.disconnect() or using EgressClient.stopEgress(). Most of our testing has been using stopEgress(). We're using HLS now and it's still happening but there's no delay in uploading the segments so it works for us. HLS is much better for us for a few reasons so we're forging ahead. Thanks again!

**Comment by biglittlebigben** [comment]
Happy to learn that HLS output works as a workaround. Would it be possible to get an idea of the rough length of the exported clip and of the file size?

**Comment by JacobSteelsmith** [comment]
@biglittlebigben When using HLS we get lots of little files as it uploads segments every 6 seconds. They are around 2.7 MB per segment as reported by S3.

**Comment by biglittlebigben** [comment]
Great, that's what I would expect. How large were the files before you switched to HLS?

**Comment by frostbyte73** [comment]
Closing this as fixed by #111 and #117
